apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -126,31 +134,36 @@ export class SendMessage {
       );
     }
 
+    const sendMessageCommand = SendMessageCommand.create({ ...command, compileContext: payload });
+
     switch (stepType) {
       case StepTypeEnum.SMS:
-        return await this.sendMessageSms.execute(command);
+        return await this.sendMessageSms.execute(sendMessageCommand);
       case StepTypeEnum.IN_APP:
-        return await this.sendMessageInApp.execute(command);
+        return await this.sendMessageInApp.execute(sendMessageCommand);
       case StepTypeEnum.EMAIL:
-        return await this.sendMessageEmail.execute(command);
+        return await this.sendMessageEmail.execute(sendMessageCommand);
       case StepTypeEnum.CHAT:
-        return await this.sendMessageChat.execute(command);
+        return await this.sendMessageChat.execute(sendMessageCommand);
       case StepTypeEnum.PUSH:
-        return await this.sendMessagePush.execute(command);
+        return await this.sendMessagePush.execute(sendMessageCommand);
       case StepTypeEnum.DIGEST:
         return await this.digest.execute(command);
       case StepTypeEnum.DELAY:
         return await this.sendMessageDelay.execute(command);
     }
   }
 
-  private async filter(command: SendMessageCommand) {
+  private async filter(command: SendMessageCommand, payload: IFilterVariables) {
     const shouldRun = await this.conditionsFilter.filter(
       ConditionsFilterCommand.create({
         filters: command.job.step.filters || [],
         environmentId: command.environmentId,
         organizationId: command.organizationId,
         userId: command.userId,
+        step: command.step,
+        job: command.job,
```

Comment:
Unrelated to this PR. It was not filtering correctly a group filter. This fixes it.